library/Fission/Web.hs

```diff
@@ -33,37 +37,59 @@ import qualified Fission.Web.User    as User
 type API = Web.Swagger.API :<|> Web.API
 
 app ::
-  ( MonadDB               m
-  , MonadTime             m
-  , MonadLogger           m
-  , MonadDNSLink          m
-  , MonadLocalIPFS        m
-  , MonadRemoteIPFS       m
-  , MonadLinkedIPFS       m
-  , MonadReflectiveServer m
+  ( MonadReflectiveServer    m
+  , MonadLinkedIPFS          m
+  , MonadRemoteIPFS          m
+  , MonadLocalIPFS           m
+  , MonadDNSLink             m
+  , MonadLogger              m
+  , MonadLogger            t
+  , MonadThrow             t
+  , MonadTime                m
+  , MonadDB                t m
+  , Heroku.AddOn.Retriever t
+  , Heroku.AddOn.Destroyer t
+  , User.Creator           t
+  , User.Retriever         t
+  , User.Modifier          t
+  , User.Destroyer         t
+  , User.CID.Creator       t
+  , User.CID.Retriever     t
+  , User.CID.Destroyer     t
   )
   => (forall a . m a -> Handler a)
   -> Context Auth.Checks
   -> Web.Host
   -> Application
-app handlerNT auth appHost = do
+app handlerNT authChecks appHost = do
   appHost
     |> server
-    |> Auth.server      api handlerNT
-    |> serveWithContext api auth
+    |> Auth.authWithContext api handlerNT
+    |> serveWithContext     api authChecks
   where
     api = Proxy @API
 
 -- | Web handlers for the 'API'
 server ::
-  ( MonadDB               m
-  , MonadTime             m
-  , MonadLogger           m
-  , MonadDNSLink          m
-  , MonadLocalIPFS        m
-  , MonadRemoteIPFS       m
-  , MonadLinkedIPFS       m
-  , MonadReflectiveServer m
+  ( MonadReflectiveServer    m
+  , MonadLinkedIPFS          m
+  , MonadRemoteIPFS          m
+  , MonadLocalIPFS           m
+  , MonadDNSLink             m
+  , MonadLogger              m
+  , MonadLogger            t
+  , MonadThrow             t
+  , MonadTime                m
+  , MonadDB                t m
+  , Heroku.AddOn.Retriever t
+  , Heroku.AddOn.Destroyer t
+  , User.Creator           t
+  , User.Retriever         t
+  , User.Modifier          t
+  , User.Destroyer         t
+  , User.CID.Creator       t
+  , User.CID.Retriever     t
+  , User.CID.Destroyer     t
```

Comment:
Check it out! We never modify CIDs or Heroku Add Ons (which totally makes sense)